docs/demo-weather-agent.md

```diff
@@ -0,0 +1,166 @@
+# Weather Agent Demo
+
```

Comment:
This doc needs a preamble pointing to the current `docs/cn-demos.md` doc to say "this is a prerequisite." 

Also can we rename cn-demos.md? The vision is all the demos start from that place no?